rdata/conversion/to_r.py

```diff
@@ -24,18 +27,31 @@
 )
 
 if TYPE_CHECKING:
-    from collections.abc import Mapping
+    from collections.abc import Callable, Mapping
     from typing import Any, Final, Literal, Protocol
 
-    Encoding = Literal["utf-8", "cp1252"]
+    import numpy.typing as npt
+
+    from rdata.unparser import FileType
 
+    Encoding = Literal["utf-8", "cp1252"]
 
     class Converter(Protocol):
-        """Protocol for Py-to-R conversion."""
+        """Protocol for class converting Python objects to R objects."""
+
+        format_version: int
+
+        def convert_to_r_sym(self, name: str) -> RObject:
+            """Convert string to R symbol."""
 
-        def __call__(self, data: Any, *, encoding: Encoding) -> RObject: # noqa: ANN401
-            """Convert Python object to R object."""
+        def convert_to_r_object(self, data: Any) -> RObject:  # noqa: ANN401
+            """Convert Python data to R object."""
 
+    ConstructorReturnValue = tuple[RObjectType, Any, dict[str, Any] | None]
```

Comment:
I am not sure if it wouldn't be better (less ambiguous and more flexible) to return the constructed R objects, even if that breaks symmetry with the reading constructors.